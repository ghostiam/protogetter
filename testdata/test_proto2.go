package testdata

import (
	"github.com/ghostiam/protogetter/testdata/proto"
)

func testInvalidProto2(t *proto.TestProto2) {
	_ = *t.D   // want `avoid direct access to proto field \*t\.D, use t\.GetD\(\) instead`
	_ = *t.F   // want `avoid direct access to proto field \*t\.F, use t\.GetF\(\) instead`
	_ = *t.I32 // want `avoid direct access to proto field \*t\.I32, use t\.GetI32\(\) instead`
	_ = *t.I64 // want `avoid direct access to proto field \*t\.I64, use t\.GetI64\(\) instead`
	_ = *t.U32 // want `avoid direct access to proto field \*t\.U32, use t\.GetU32\(\) instead`
	_ = *t.U64 // want `avoid direct access to proto field \*t\.U64, use t\.GetU64\(\) instead`
	_ = *t.T   // want `avoid direct access to proto field \*t\.T, use t\.GetT\(\) instead`
	_ = t.B    // want `avoid direct access to proto field t\.B, use t\.GetB\(\) instead`
	_ = *t.S   // want `avoid direct access to proto field \*t\.S, use t\.GetS\(\) instead`
}
