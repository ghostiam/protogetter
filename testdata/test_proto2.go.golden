package testdata

import (
	"github.com/ghostiam/protogetter/testdata/proto"
)

func testInvalidProto2(t *proto.TestProto2) {
	_ = t.GetD()   // want `avoid direct access to proto field \*t\.D, use t\.GetD\(\) instead`
	_ = t.GetF()   // want `avoid direct access to proto field \*t\.F, use t\.GetF\(\) instead`
	_ = t.GetI32() // want `avoid direct access to proto field \*t\.I32, use t\.GetI32\(\) instead`
	_ = t.GetI64() // want `avoid direct access to proto field \*t\.I64, use t\.GetI64\(\) instead`
	_ = t.GetU32() // want `avoid direct access to proto field \*t\.U32, use t\.GetU32\(\) instead`
	_ = t.GetU64() // want `avoid direct access to proto field \*t\.U64, use t\.GetU64\(\) instead`
	_ = t.GetT()   // want `avoid direct access to proto field \*t\.T, use t\.GetT\(\) instead`
	_ = t.GetB()   // want `avoid direct access to proto field t\.B, use t\.GetB\(\) instead`
	_ = t.GetS()   // want `avoid direct access to proto field \*t\.S, use t\.GetS\(\) instead`
}
